crates/nu-utils/standard_library/std.nu

```diff
@@ -26,51 +32,182 @@ def _assertion-error [start, end, label, message?: string] {
+            text: "It is not true.",
+            start: (metadata $condition).span.start,
+            end: (metadata $condition).span.end
+        })
+    }
 }
 
-# ```nushell
-# ❯ assert eq 3 3
-# ❯ assert eq 3 1
-# Error:
-#   × Assertion failed.
-#    ╭─[entry #14:1:1]
-#  1 │ assert eq 3 1
-#    ·           ─┬─
-#    ·            ╰── They are not equal: 3 != 1
-#    ╰────
+# Assert that executing the code generates an error
 #
+# For more documentation see the assert command
+# 
+# # Examples
 #
-# ```
-export def "assert eq" [left: any, right: any, message?: string] {
-    let left_start = (metadata $left).span.start
-    let right_end = (metadata $right).span.end
+# > assert error {|| missing_command} # passes
+# > assert error {|| 12} # fails
+export def "assert error" [
+    code: closure,
+    message?: string
+] {
+    let error_raised = (try { do $code; false } catch { true })
+    assert ($error_raised) $message --error-details {
+        start: (metadata $code).span.start
+        end: (metadata $code).span.end
+        text: $"There were no error during code execution: (view source $code)"
+    }
+}
 
-    if ($left != $right) {
-        _assertion-error $left_start $right_end $"They are not equal: ($left) != ($right)" $message
+# Assert $left == $right
+#
+# For more documentation see the assert command
+#
+# # Examples
+# 
+# > assert equal 1 1 # passes
+# > assert equal 1 2 # fails
+export def "assert equal" [left: any, right: any, message?: string] {
+    assert ($left == $right) $message --error-details {
+        start: (metadata $left).span.start
+        end: (metadata $right).span.end
+        text: $"They are not equal. Left = ($left). Right = ($right)."
     }
 }
 
-# ```nushell
-# ❯ assert ne 1 3
-# ❯ assert ne 42 42
-# Error:
-#   × Assertion failed.
-#    ╭─[entry #23:1:1]
-#  1 │ assert ne 42 42
-#    ·           ──┬──
-#    ·             ╰── They both are 42
-#    ╰────
-# ```
-export def "assert ne" [left: any, right: any, message?: string] {
-    let left_start = (metadata $left).span.start
-    let right_end = (metadata $right).span.end
+# Assert $left != $right
+#
+# For more documentation see the assert command
+#
+# # Examples
+#
+# > assert not equal 1 2 # passes
+# > assert not equal 1 "apple" # passes
+# > assert not equal 7 7 # fails
+export def "assert not equal" [left: any, right: any, message?: string] {
+    assert ($left != $right) $message --error-details {
+        start: (metadata $left).span.start
+        end: (metadata $right).span.end
+        text: $"They both are ($left)."
+    }
+}
 
-    if ($left == $right) {
-        _assertion-error $left_start $right_end $"They both are ($left)" $message
+# Assert $left <= $right
+#
+# For more documentation see the assert command
+#
+# # Examples
+#
+# > assert less or equal 1 2 # passes
+# > assert less or equal 1 1 # passes
+# > assert less or equal 1 0 # fails
+export def "assert less or equal" [left: any, right: any, message?: string] {
```

Comment:
for `less or equal`, `less`, `greater` and `greater or equal`, maybe you should annotate the `left` and `right` arguments with `number` instead of `any`.
it looks we can only compare integers, floats, hexadecimal values, ...
these can all be annotated with the `number` type :+1:

Reply:
Numbers are not comparable:
```
  × Types mismatched for operation.
     ╭─[std.nu:112:1]
 112 │ export def "assert less or equal" [left: number, right: number, message?: string] {
 113 │     assert ($left <= $right) $message --error-label {
     ·             ──┬── ─┬ ───┬──
     ·               │    │    ╰── number
     ·               │    ╰── doesn't support these values.
     ·               ╰── number
 114 │         start: (metadata $left).span.start
     ╰────
  help: Change number or number to be the right types and try again.
```